packages/graphql-migrations/src/util/isAutoIncrementableColumn.ts

```diff
@@ -0,0 +1,21 @@
+import Knex from 'knex';
+
+/**
+ * Detect if the column is incrementable based on column information
+ * 
+ * @param {string} client - database client 
+ * @param {Knex.ColumnInfo} columnInfo - metadata for column
+ */
+export function isAutoIncrementableColumn (client: string, columnInfo: Knex.ColumnInfo): boolean {
+  switch (client) {
+    case 'pg':
+      return columnInfo.defaultValue?.toString().startsWith('nextval(');
```

Comment:
Should we check if the type is `integer` or `bigInteger` too? Because a `varchar` with a default value of `'nextval(dshdjhsjdh)'` whould be auto-incrementable too with this logic.

Reply:
My reasoning for not checking it is because this is what is being read from the remote database, and if the default value is `nextval('whatever')`  that is an error on the remote database and it should not handed by Postgres:

```sql
create table IF NOT EXISTS testtable (
	column1 varchar default nextval('xxxxx'),
	secondcol text
);
```

This throws:

```
ERROR:  relation "dshdjhsjdh" does not exist
SQL state: 42P01
```

I have also been able to successfully create a sequence and auto-incrementing logic on a varchar.

You can try it out here and see the auto-incrementing in action:

```sql
CREATE SEQUENCE if not exists testtable_id_seq;

create table IF NOT EXISTS testtable (
	column1 varchar default nextval("'testtable_id_seq'"),
	secondcol text
);

insert into testtable (secondcol) values ('hello');
```

The `testtable_id_seq` sequence is decoupled from the datatypes which use it, and "1", "2"... are also valid varchars so it works just fine.

Reply:
Okay, thanks for the findings and great explanation. My reason was highly biased on `ID`as `integer`.  Sorry for the noise.